GetDKAN/dkan
# Dataset collections - initial build-out

Connects isdapps/adc-code#1588
Relates to #2219
Relates to #2137 

This PR introduces collections support for datasets. 
**Functionality Implemented:**

- Search row support for collections
- Parent Child association on node view (listed links, with see more link to solr search page)
- Collections Faceted search
- Child parent association on node view.

## Note
- Creating search index fields and facets doesn't seem to be (easily) doable, so this mr only has collections fields and facet logic applied to datasets index. Once dkan_sci_metadata is integrated in dkan codebase, we will need to add collections field logic to the primary search index contained within = dkan_sci_search, @dafeder 

## QA Steps
- [x] Visit node/add/dataset and verify presence of Collection autocomplete field.
- [x] Choose 11 published dataset nodes, and edit them by adding the same dataset ref in the collection field to all 11
- [x] Visually verify on child nodes that a singular collection parent link appears on the node view
- [x] Visually verify on the parent node view that a dataset view of three datasets (all from the children ref'd) is listed, with a see more link, below
- [ ] Click the see more link from parent node and visually verify it takes you to a search page with only items from the parent collection, with a collection facet opened and listing the current parent dataset.
- [ ] Search for a child ite's title in the above collection on the search page after clearing all filters.

* Steps below should only be applied when we implement the dkan_sci_index - datasets index has old search view config. For now, these steps are not necessary.
- [ ] Visually verify that the parent node is listed in the view row for the child item
- [ ] Visually verify that datasets that are not associated with a collection do not show a collection label in the view row.
